### ISSUE TITLE
Add option to always generate platform specific folders

### ISSUE THREAD

**Issue by vRallev** [Kotlin/binary-compatibility-validator#174]
In our Kotlin Multiplatform project we use two JVM targets (Android and desktop), therefore this plugin will use two different folders for the API dumps: `api/android` and `api/desktop`. However, in CI we disable the desktop target and only Android is present. The `apiCheck` then will fail, because it doesn't look in the right directory for the dump. The code is [here](https://github.com/Kotlin/binary-compatibility-validator/blob/master/src/main/kotlin/BinaryCompatibilityValidatorPlugin.kt#L80).

To avoid the issue, either an option to always us the platform specific folder no matter how many JVM targets would work, or as an alternative an option to disable targets, e.g. we'd disable desktop and only use Android.

**Comment by fzhinkin** [comment]
@vRallev how exactly do you disable the second target for the CI? Or maybe you have a small reproducer already?

**Comment by fzhinkin** [comment]
Thanks for the reproducer!

> To avoid the issue, either an option to always us the platform specific folder no matter how many JVM targets would work, or as an alternative an option to disable targets, e.g. we'd disable desktop and only use Android.

Just to explore possible solutions: would an option to use flat dump files layout (like `api/<moduleName>.android.api`, `api/<moduleName>.desktop.api`) work for you?

**Comment by vRallev** [comment]
Yes, it would. The important piece here is that the output path and file name must stay consistent and not depend on how many JVM targets are there, so it shouldn't switch between `<moduleName>.android.api` and `<moduleName>.api`.

**Comment by shanshin** [comment]
As part of [the migration of separate BCV functionality to Kotlin Gradle Plugin](https://youtrack.jetbrains.com/issue/KT-71098/Migrate-Binary-Compatibility-Validator-to-Kotlin-Gradle-Plugin), the addition of new features to a separate plugin has been discontinued.

It will be developed a format that eliminates this problem there: https://youtrack.jetbrains.com/issue/KT-78009
